docs/architecture/adr-021-subspace-specific-custom-fee-tokens.md

```diff
@@ -52,82 +42,51 @@ message Subspace {
-2. update `x/subspaces` parameters by governance.
+We will implement the operation that allows subspace admins to updates subspace allowed fee tokens list by governance;
 
 ```proto
 service Msg {
-  // UpdateSubspaceFeeTokens allows subspace managers to update the allowed tokens to be fee tokens inside the subspace
+  // UpdateSubspaceFeeTokens allows subspace admins to update the allowed tokens to be fee tokens inside the subspace by governance
   rpc UpdateSubspaceFeeTokens(MsgUpdateSubspaceFeeTokens) returns (MsgUpdateSubspaceFeeTokensResponse);
-    
-  // UpdateParams defines a (governance) operation for updating the module
-  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
 }
 
-// MsgUpdateSubspaceFeeTokens represents the message to be used to update a subspace fee tokens
+// MsgUpdateSubspaceFeeTokens represents the message to be used to update a subspace fee tokens by governance
 message MsgUpdateSubspaceFeeTokens {
 
   // Id of the subspace where the list of allowed fee tokens will be updated
   uint64 subspace_id = 1;
     
-  // List of the allowed tokens to be fee token inside the subspace
-  repeated string allowed_fee_tokens = 2;
+  // List of the allowed tokens to be fee token inside the subspace along with their default minimum prices
+  repeated Coin allowed_fee_tokens = 2;
     
-  // Address of the sender/signer
-  string signer = 3;
+  // authority is the address that controls the module (defaults to x/gov unless overwritten).
+  string authority = 3;
 }
 
 // MsgUpdateSubspaceFeeTokensResponse represents the Msg/UpdateSubspaceFeeTokens response type
 message MsgUpdateSubspaceFeeTokensResponse {}
-
-
-// MsgUpdateParams is the Msg/UpdateParams request type.
-message MsgUpdateParams {
-  string authority = 1;
-  Params params = 2;
-}
-
-// MsgUpdateParamsResponse defines the response structure for executing a
-// MsgUpdateParams message.
-message MsgUpdateParamsResponse {}
 ```
 
-### Query
-
-We will also provide a new query endpoint that enables users to check `x/subspaces` parameters.
-
-```proto
-service Query {
-  // Params allows to query the module parameters
-  rpc Params(QueryParamsRequest) returns (QueryParamsResponses) {
-    option (google.api.http).get = "/desmos/subspaces/v3/params";
-  };
-}
+### Custom TxFeeChecker
 
-// QueryParamsRequest is the request type for Query/Params RPC method
-message QueryParamsRequest {}
+To make it easier for validators to manage the minimum prices of fee tokens allowed within a subspace, we will develop a new `TxFeeChecker` based on the existing one, which will function as follows:
+- Combine the list of minimum gas prices in the validator's local configuration with the list of allowed fee tokens and their minimum prices within the subspace.
+- Follow the same process as the existing `TxFeeChecker`.
 
-// QueryParamsResponse is the response type for Query/Params RPC method
-message QueryParamsResponse {
-  Params params = 1;
-}
-```
+### Update `x/fees`
```

Comment:
Since `x/fees` may conflict with subspace fee token checks, I think we can update the logic of `CheckFees`. What do you think? @RiccardoM